File: Scripts/results/generated_code_results/Flutter/bak_code/News&Information/MSNWeather/SettingsScreen.dart
import 'package:flutter/material.dart';
class SettingsScreen extends StatefulWidget {
const SettingsScreen({Key? key}) : super(key: key);
@override
_SettingsScreenState createState() => _SettingsScreenState();
}
class _SettingsScreenState extends State<SettingsScreen> {
int _selectedIndex = 0;
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pushNamed(context, 'PersonalCentral'),
),
title: Text('Settings'),
actions: [
Container(
width: 40,
height: 40,
decoration: BoxDecoration(
shape: BoxShape.circle,
color: Colors.black,
),
),
],
),
body: ListView(
children: [
_buildSectionHeader('Accounts'),
_buildAccountItem('Microsoft account', Icons.person),
_buildAccountItem('Work or school account', Icons.work),
_buildSectionHeader('General'),
_buildSettingsItem('Privacy', Icons.security),
_buildSettingsItem('Region and language', Icons.language),
_buildSettingsItem('Permissions', Icons.lock),
_buildSectionHeader('Preferences'),
_buildSettingsItem('Homepage', Icons.home),
_buildSettingsItem('Notifications', Icons.notifications),
_buildSettingsItem('Theme', Icons.palette),
_buildSettingsItem('Search', Icons.search),
_buildSettingsItem('Rewards', Icons.card_giftcard),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: _selectedIndex,
onTap: (index) {
setState(() {
_selectedIndex = index;
});
if (index == 0) {
Navigator.pushNamed(context, 'Home');
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.cloud), label: 'Weather'),
BottomNavigationBarItem(icon: Icon(Icons.chat), label: 'Chat'),
BottomNavigationBarItem(
icon: Stack(
children: [
Icon(Icons.tab),
Positioned(
right: 0,
child: Container(
padding: EdgeInsets.all(1),
decoration: BoxDecoration(
color: Colors.red,
borderRadius: BorderRadius.circular(6),
),
constraints: BoxConstraints(
minWidth: 12,
minHeight: 12,
),
child: Text(
'4',
style: TextStyle(
color: Colors.white,
fontSize: 8,
),
textAlign: TextAlign.center,
),
),
),
],
),
label: 'Tabs',
),
BottomNavigationBarItem(icon: Icon(Icons.apps), label: 'Apps'),
],
),
);
}
Widget _buildSectionHeader(String title) {
return Padding(
padding: const EdgeInsets.fromLTRB(16, 16, 16, 8),
child: Text(
title,
style: TextStyle(
fontSize: 14,
fontWeight: FontWeight.bold,
color: Colors.grey[600],
),
),
);
}
Widget _buildAccountItem(String title, IconData icon) {
return ListTile(
leading: Icon(icon),
title: Text(title),
trailing: TextButton(
child: Text('Sign in'),
onPressed: () {
// Handle sign in action
},
),
);
}
Widget _buildSettingsItem(String title, IconData icon) {
return ListTile(
leading: Icon(icon),
title: Text(title),
onTap: () {
// Handle settings item tap
},
);
}
}

File: Scripts/results/generated_code_results/baseline_claude/News&Information/MSNWeather/SettingsScreen.js
import React from 'react';

import { View, Text, Image, TouchableOpacity, StyleSheet, SafeAreaView, ScrollView } from 'react-native';

import { useNavigation } from '@react-navigation/native';



const SettingsScreen = () => {

  const navigation = useNavigation();



  const settingsItems = [

    { section: 'Accounts', items: [

      { icon: 'microsoft', title: 'Microsoft account', action: 'Sign in' },

      { icon: 'work', title: 'Work or school account', action: 'Sign in' },

    ]},

    { section: 'General', items: [

      { icon: 'privacy', title: 'Privacy' },

      { icon: 'globe', title: 'Region and language' },

      { icon: 'permissions', title: 'Permissions' },

    ]},

    { section: 'Preferences', items: [

      { icon: 'home', title: 'Homepage' },

      { icon: 'bell', title: 'Notifications' },

      { icon: 'theme', title: 'Theme' },

      { icon: 'search', title: 'Search' },

      { icon: 'rewards', title: 'Rewards' },

    ]},

  ];



  const renderSettingsItem = (item, index) => (

    <TouchableOpacity key={index} style={styles.settingsItem}>

      <Image source={{ uri: `../assets/snack-icon.png` }} style={styles.itemIcon} />

      <Text style={styles.itemTitle}>{item.title}</Text>

      {item.action && <Text style={styles.itemAction}>{item.action}</Text>}

    </TouchableOpacity>

  );



  return (

    <SafeAreaView style={styles.container}>

      <View style={styles.header}>

        <TouchableOpacity onPress={() => navigation.goBack()}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.backIcon} />

        </TouchableOpacity>

        <Text style={styles.headerTitle}>Settings</Text>

      </View>



      <ScrollView>

        {settingsItems.map((section, sectionIndex) => (

          <View key={sectionIndex} style={styles.section}>

            <Text style={styles.sectionTitle}>{section.section}</Text>

            {section.items.map(renderSettingsItem)}

          </View>

        ))}

      </ScrollView>



      <View style={styles.bottomNav}>

        <TouchableOpacity style={styles.navItem} onPress={() => navigation.navigate('Home')}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.navIcon} />

          <Text style={styles.navText}>Home</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem} onPress={() => navigation.navigate('Weather')}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.navIcon} />

          <Text style={styles.navText}>Weather</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.navIcon} />

          <Text style={styles.navText}>Chat</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.navIcon} />

          <Text style={styles.navText}>Tabs</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem} onPress={() => navigation.navigate('Apps')}>

          <Image source={{ uri: '../assets/snack-icon.png' }} style={styles.navIcon} />

          <Text style={styles.navText}>Apps</Text>

        </TouchableOpacity>

      </View>

    </SafeAreaView>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#F0F0F5',

  },

  header: {

    flexDirection: 'row',

    alignItems: 'center',

    padding: 15,

    backgroundColor: '#FFFFFF',

    borderBottomWidth: 1,

    borderBottomColor: '#E0E0E0',

  },

  backIcon: {

    width: 24,

    height: 24,

    marginRight: 15,

  },

  headerTitle: {

    fontSize: 20,

    fontWeight: 'bold',

    color: '#4285F4',

  },

  section: {

    marginBottom: 20,

  },

  sectionTitle: {

    fontSize: 16,

    fontWeight: 'bold',

    color: '#666',

    padding: 15,

  },

  settingsItem: {

    flexDirection: 'row',

    alignItems: 'center',

    padding: 15,

    backgroundColor: '#FFFFFF',

    borderBottomWidth: 1,

    borderBottomColor: '#E0E0E0',

  },

  itemIcon: {

    width: 24,

    height: 24,

    marginRight: 15,

  },

  itemTitle: {

    flex: 1,

    fontSize: 16,

  },

  itemAction: {

    color: '#4285F4',

    fontSize: 14,

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    backgroundColor: '#FFFFFF',

    paddingVertical: 10,

    borderTopWidth: 1,

    borderTopColor: '#E0E0E0',

  },

  navItem: {

    alignItems: 'center',

  },

  navIcon: {

    width: 24,

    height: 24,

    marginBottom: 5,

  },

  navText: {

    fontSize: 12,

    color: '#666',

  },

});



export default SettingsScreen;